tritium-lib/src/main/java/com/palantir/tritium/proxy/InvocationEventProxy.java

```diff
@@ -185,11 +191,11 @@ private static void logInvocationWarningOnFailure(@Nullable InvocationContext co
 
     static void logInvocationWarning(String event, Object instance, Method method, Throwable cause) {
         if (logger.isWarnEnabled()) {
-            logger.warn("{} exception handling {} invocation of {}.{} on {}",
+            logger.warn("{} exception handling {} invocation of {} {} on {}",
                     safeSimpleClassName("throwable", cause),
                     SafeArg.of("event", event),
-                    SafeArg.of("class", method.getDeclaringClass()),
-                    SafeArg.of("method", method.getName()),
+                    SafeArg.of("class", method.getDeclaringClass().getName()),
+                    SafeArg.of("method", method),
```

Comment:
+1 passing the method object here, since method.toString provides more information.

Should the value of `class` use `instance.getClass() /* maybe .getName() */` instead of the method declaring class to provide more information describing the implementation?

Reply:
will tweak this in a followup commit that does a bunch of random cleanup and some JIT optimizations